src/dotnet/commands/dotnet-cache/Program.cs

```diff
@@ -0,0 +1,85 @@
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            CommandLineApplication app = new CommandLineApplication(throwOnUnexpectedArg: false);
+            app.Name = "dotnet cache";
+            app.FullName = LocalizableStrings.AppFullName;
+            app.Description = LocalizableStrings.AppDescription;
+            app.AllowArgumentSeparator = true;
+            app.ArgumentSeparatorHelpText = HelpMessageStrings.MSBuildAdditionalArgsHelpText;
+            app.HelpOption("-h|--help");
+
+            CommandOption projectArgument = app.Option(
+                $"-e|--entries <{LocalizableStrings.ProjectEntries}>", LocalizableStrings.ProjectEntryDescription,
+                CommandOptionType.SingleValue);
+
+            CommandOption frameworkOption = app.Option(
+                $"-f|--framework <{LocalizableStrings.FrameworkOption}>", LocalizableStrings.FrameworkOptionDescription,
+                CommandOptionType.SingleValue);
+
+            CommandOption runtimeOption = app.Option(
+                $"-r|--runtime <{LocalizableStrings.RuntimeOption}>", LocalizableStrings.RuntimeOptionDescription,
+                CommandOptionType.SingleValue);
+
+            CommandOption outputOption = app.Option(
+                $"-o|--output <{LocalizableStrings.OutputOption}>", LocalizableStrings.OutputOptionDescription,
+                CommandOptionType.SingleValue);
+
+            CommandOption fxOption = app.Option(
+                $"--framework-version <{LocalizableStrings.FrameworkVersionOption}>", LocalizableStrings.FrameworkVersionOptionDescription,
+                CommandOptionType.SingleValue);
+
+            CommandOption optimizationOption = app.Option(
+                $"--skip-optimization <{LocalizableStrings.SkipOptmizationOption}>", LocalizableStrings.SkipOptmizationOptionDescription,
+                CommandOptionType.SingleValue);
+
+            CommandOption workingDir = app.Option(
+               $"-w |--working-dir <{LocalizableStrings.IntermediateWorkingDirOption}>", LocalizableStrings.IntermediateWorkingDirOptionDescription,
+               CommandOptionType.SingleValue);
+
+            CommandOption preserveWorkingDir = app.Option(
+               $"--preserve-working-dir <{LocalizableStrings.PreserveIntermediateWorkingDirOption}>", LocalizableStrings.PreserveIntermediateWorkingDirOptionDescription,
+               CommandOptionType.SingleValue);
+
+            CommandOption verbosityOption = MSBuildForwardingApp.AddVerbosityOption(app);
+
+            app.OnExecute(() =>
+            {
+
+                var cache = new CacheCommand();
+
+                cache.Framework = frameworkOption.Value();
+                cache.Runtime = runtimeOption.Value();
+                cache.OutputPath = outputOption.Value();
+                cache.FrameworkVersion = fxOption.Value();
+                cache.Verbosity = verbosityOption.Value();
+                bool skipOptimization;
```

Comment:
If you do what I suggested above, you won't have to do this TryParse piece here.